samples/sdl_joystick_event/main.c

```diff
@@ -0,0 +1,63 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+#define ARRAY_SIZE(x) (sizeof(x)/sizeof(x[0]))
+
+int main() {
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_JOYSTICK);
+
+  int joystickIndex = 0;
+  SDL_Joystick *joy = SDL_JoystickOpen(joystickIndex);
+  if (joy == NULL) {
+    debugPrint("Could not open Joystick %i, exiting...", joystickIndex);
+    SDL_QuitSubSystem(SDL_INIT_JOYSTICK);
+    Xsleep(2000);
+    return -1;
+  }
+
+  const char* buttonNames[10] = {"A", "B", "X", "Y", "White", "Black", "Back", "Start", "LeftThumb", "RightThumb"};
+  const char* axisNames[6] = {"LX", "LY", "LTrigg", "RX", "RY", "RTrigg"};
+
+  debugPrint("Num joysticks    : %02i\n", SDL_NumJoysticks());
+  debugPrint("Joystick name    : %s\n", SDL_JoystickNameForIndex(joystickIndex));
```

Comment:
There's an API to get name by index, and another API to get it by opened handle.
At this point you already own a handle.